Sprint 8, 3 HW

vaiwork opened the pull request.

vaiwork committed:
In activity_main.xml create my first activity.

vaiwork committed:
Homework 1 from Sprint 8 Theme 2

vaiwork committed:
Homework 2 from Sprint 8 Theme 3.
Added click listeners for buttons on main activity:
1. Added click listener for search button with anonimous class
2. Added click listeners for media and settings buttons with lambda statement

vaiwork committed:
Homework 3 from Sprint 8 Theme 4
Added navigation from Main activity to Search, Media and Settings activities.

vaiwork committed:
Delete .gitignore

SeveNChaK commented:
К слову, не надо добавлять папку `.idea` к отслеживанию в `git`. Она у каждого своя и генерируется при создании/клонирование проекта на устройстве.

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,112 @@
+    android:orientation="vertical"
+    >
+
+    <LinearLayout
```
⚠️ Кнопка назад не выглядит так, как показано в макете Figma. Сейчас она выглядит как кнопка с тенями и не приводит ни к каким действиям.

⏫ Стоит ее сделать работающей. Так как ты решил делать кастомный тулбар, то лучше всего будет использовать `Toolbar` view. В него можно положить любую верстку. Для этого экрана вижу это так:
```
<androidx.appcompat.widget.Toolbar
    android:id="@+id/settings_toolbar"
    android:layout_width="match_parent"
    android:layout_height="?attr/actionBarSize"
    app:navigationIcon="@drawable/arrow_back_btn"
    android:layout_marginBottom="24dp">

    <TextView
            android:layout_width="wrap_content"
            android:layout_height="match_parent"
            android:text = "@string/settings"
            android:textColor="@color/button_text"
            android:textSize="@dimen/textSize" 
            android:textStyle="bold"
            android:gravity="center_vertical" />

</androidx.appcompat.widget.Toolbar>
```
И еще придется дописать в активити в методе `onCreate` немного кода:
```
 override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_settings)
     val toolbar = findViewById<Toolbar>(R.id.settings_toolbar)
     toolbar.setNavigationOnClickListener { onBackPressed() }
 }
```
Нужно найти наш `Toolbar` и сказать, что по клику на иконку, мы делаем действие "назад".

vaiwork replied:
В задании ничего не сказано про работающую кнопку. Поэтому она и не выполнет никаких действий. Даже про переходы на экраны написано, что переходы на экраны должны быть с главных экранов на другие (прямо так и написано), а вот про кнопку возвращения ничего не написано.

SeveNChaK replied:
Поэтому пункт про работающую кнопку помечен как улучшение. Главным замечанием было, что кнопка не выглядит как в макете.

SeveNChaK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,112 @@
+        android:gravity="top"
+        />
+
+    <Switch
```
⚠️ Данный пункт и пункты ниже в макете Figma не являются переключателями. Сейчас по клику на них двигаются иконки.

vaiwork replied:
По поводу переключателей. В домашнем задании сказано следующее:
"Для отрисовки «переключателя» на экране настроек примените класс Switch."
Я понимаю, что сейчас в Figma нет переключателя, однако, в задании написано было сделать так, поэтому получилась ситуация, в которой нужно было выбирать одно из двух. Был выбран вариант с переключателем, потому что в Figma он всё же есть, но я так понимаю не активный. Поэтому по данному пункту тоже спорная ситуация. Как и с другими ниже. Грубо говоря всё упирается именно в кретерии выполнения задания, поэтому я пока не понимаю где я сделал не так как было написано в заданиях или отрисовано в Figma.

SeveNChaK replied:
В макете переключателем является пункт с темой. Для него и дается подсказка, как стоит реализовать его. Все остальное - не переключатели.

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,122 @@
+
+            android:layout_width="match_parent"
+            android:layout_height="0dp"
+            android:layout_marginBottom="@dimen/buttonBetween"
```
⚠️ Отступы между кнопка не соответствуют макету в Figma. Из-за того, что ты указываешь отступ здесь снизу и у следующей кнопки сверху, получается, что отступ становится в 2 раза больше необходимого.

vaiwork replied:
Согласен, действительно два раза получался отступ между кнопками. Будет переделано.

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,122 @@
+            android:text="@string/settings"
+            android:textColor="@color/button_text"
+            android:textSize="@dimen/textSize"
+            android:textAllCaps="false"/>
```
⏫ В этом аттрибуте нет необходимости, так как все ресуры и так созданы у тебя без капса.

vaiwork replied:
Если я уберу данный аттрибут, то все буквы будут заглавные, а в Figma написано, что только первая буква должна быть заглавная. Поэтому без данного аттрибута моя верстка не будет соответствовать верстке в Figma.

SeveNChaK replied:
Согласен. `Button` сам по себе делает капсом весь текст.

SeveNChaK commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,122 @@
+        android:layout_marginStart="4dp"
+        android:layout_marginTop="4dp"
+        >
+        <TextView
```
⏫ В этой вложенности нет необходимости. Можно `TextView` вынести из `FrameLayout`.

vaiwork replied:
В Figma если посмотреть, то TextView находится в RelativeLayout, рядом с которой какаято кнопка поиска. А в свою очередь кнопка поиска и Layout находтся в другом RelativeLayout. Если я вынесу TextView, то в данном случае моя верстка не будет соответствовать Figma.

SeveNChaK replied:
Вложенность элементов в Figma может быть обусловлена особенностью сервиса и видением человека, который этот макет делал. Однако, когда макет переносится в приложение, стоит учитывать необходимость этой вложености в том месте, где реализовывается макет. В нашем случае это верста лейаута под Android. Потому что в системе могут быть свои особенности взаимодействи и реализации, и не всегда можно будет сделать точную копию структуры. То, что в макете есть какие-то невидимые элементы, для нас особой роли не играет, так как в задание никакого взаимодействия с ними нет. А под словами "соответствовать макету в Figma" понимается визуальная состовляющая, а не точная копия стукртуры.

SeveNChaK commented on app/src/main/java/com/vaiwork/playlistmaker/SetingsActivity.kt:
```diff
@@ -0,0 +1,11 @@
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+
+class SetingsActivity : AppCompatActivity() {
```
Опечатка Se**tt**ings =)

SeveNChaK reviewed (commented):
В некоторых файлах код прыгает, отступы не одинаковые - рекомендую использовать встроенные форматтер. Если ты программируешь в IDE от JetBrains, то можешь пройтись по всем файлам и нажать комбинацию (**Windows**: `Shift + Ctrl + Alt + L`, **MacOS**: `Shift + Option + Command + L`), затем выбрать, какую часть кода хочешь изменить, и выполнить команду. Среда разработки автоматически сделает нужные отступы и переносы (как она это делает можно изменить в настройках, но дефолтные значения вполне нормальные).

vaiwork committed:
Changes after review.
1. Edited margins between buttons on main activity.
2. Changed 3 switches to LinearLayout with TextView and ImageView.
I didn't touch first switch (dark mode) because this switch active on Figma
3. Added action to back toolbar on settings activity.
